path-manager/app/services/PathStore.scala

```diff
@@ -96,7 +95,75 @@ object PathStore {
     }
   }
 
-  def updateCanonical(newPath: String, id: Long) = {
+  def registerAlias(proposedAliasPathRecord: PathRecord) = {
+
+    val id = proposedAliasPathRecord.identifier
+
+    Logger.debug(s"Registering path [${proposedAliasPathRecord.path }] for [$id]")
+
+    if (!PathValidator.isValid(proposedAliasPathRecord.path)) {
+      Left(s"invalid path [${proposedAliasPathRecord.path}]")
+    } else {
+      val existingPath = Option(Dynamo.pathsTable.getItem("path", proposedAliasPathRecord.path)).map(PathRecord(_))
+      val index: Index = Dynamo.pathsTable.getIndex("id-index")
+
+      existingPath match {
+        case Some(pr) if (pr.identifier != id) => {
+          Logger.warn(s"Failed to register existing path [${proposedAliasPathRecord.path}], already claimed by id [${pr.identifier}], submitting id [$id]")
+          Left("path already in use")
+        }
+        case _ => {
+          val shortUrlPathRecord = PathRecord(ShortUrlEncoder.generateShortUrlPath(id), id, "short", proposedAliasPathRecord.system)
```

Comment:
are we generating another 'short' url here for the alias? could it not re-use the short URL for the canonical, is that not effectively already an alias (just with a generated path)

Reply:
Good spot, should have deleted after copy.